x-pack/plugins/cases/public/common/hooks.ts

```diff
@@ -5,11 +5,120 @@
  * 2.0.
  */
 
+import { useCallback, useEffect, useState } from 'react';
+
+import { useLocation, useHistory } from 'react-router-dom';
+
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+import { parse, stringify } from 'query-string';
 import { STACK_APP_ID } from '../../common/constants';
 import { useCasesContext } from '../components/cases_context/use_cases_context';
 
+import { DEFAULT_QUERY_PARAMS } from '../containers/use_get_cases';
+
+import type {
+  LocalStorageQueryParams,
+  QueryParams,
+  ParsedUrlQueryParams,
+  UrlQueryParams,
+} from '../../common/ui/types';
+
+const parseUrlQueryParams = (search: string): UrlQueryParams => {
+  const parsedUrlParams: ParsedUrlQueryParams = parse(search);
+
+  const urlParams: UrlQueryParams = {
+    ...(parsedUrlParams.sortField && { sortField: parsedUrlParams.sortField }),
+    ...(parsedUrlParams.sortOrder && { sortOrder: parsedUrlParams.sortOrder }),
+  };
+
+  const intPage = parsedUrlParams.page && parseInt(parsedUrlParams.page, 10);
+  const intPerPage = parsedUrlParams.perPage && parseInt(parsedUrlParams.perPage, 10);
+
+  if (intPage) {
```

Comment:
If the `parsedUrlParams.page` is set to `"0"` this if block won't get executed right? Is that what we want?

Reply:
Good catch.

I think luckily the outcome would be ok. Zero is an invalid page so ignoring it and reverting to the previous value is fine. 

Should I still change the page it if is `0`?

Reply:
Ah good point, na you can leave it then. I'd just put in a comment stating that page=0 is invalid and this code would ignore it.